### ISSUE TITLE
OBJBLOCK and ENUM_CONSTANT_DEF both match enum constants

### ISSUE THREAD

**Issue by leventov** [checkstyle/checkstyle#4738]
When e. g. checking LeftCurly, it's impossible to configure different policy for OBJBLOCK and ENUM_CONSTANT_DEF, because they both match enum constants, that should be matched only by ENUM_CONSTANT_DEF, but not by OBJBLOCK.

**Comment by romani** [comment]
issue has to be formatted as requested - http://checkstyle.sourceforge.net/report_issue.html#How_to_report_a_bug

incorrectly reported issue will be closed in a month.

**Comment by leventov** [comment]
It's a feature request, not a bug. OBJBLOCK token type covers many constructs, including anonymous class expressions and enum constant definitions. There is no way in Checkstyle right now to configure LeftCurly so that code formatted like this passes the checks, and anonymous class expressions' new line style is checked:
```java
enum Foo
{
  BAR { // enum - LeftCurly eol
  };

  void foo()
  {
    new Thread()
    { // anonymous class - LeftCurly new line
      public void run()
      {
      }
    }.run();
  }
}
```

Possible solution is to add `ANON_CLASS_DEF` token type.

**Comment by romani** [comment]
http://checkstyle.sourceforge.net/config_blocks.html#LeftCurly is simple, and limited by design.
You need a bit advanced Check to keep your policies in consideration to diff curly braces of Enum, Class, Anonymous class, ..... . You  need to write your own Check for this.
If you do not have time to create you own Check, you can share design ideas here and may be smb from community can pick up the issue. 

You need to make a description of issue (design proposal) clear with number of usecases examples, with all properties of Check that might be required. You need to imagine that Check already exists and you just describe behavior of it. So there should be imaginary xml config examples and small pieces of code. As you did a design, I need to review it , and approve it. After issue got label "approved" work could be started and have very good chance to be merged.
